atom/browser/api/atom_api_app.cc

```diff
@@ -912,6 +924,47 @@ void App::GetFileIcon(const base::FilePath& path,
   }
 }
 
+v8::Local<v8::Value> App::GetAppMemoryInfo(v8::Isolate* isolate) {
+  AppIdProcessIterator process_iterator;
+  auto processEntry = process_iterator.NextProcessEntry();
+  std::vector<mate::Dictionary> result;
+
+  while (processEntry != nullptr) {
+    int64_t pid = processEntry->pid();
+    auto process = base::Process::OpenWithExtraPrivileges(pid);
+
+#if defined(OS_MACOSX)
+    std::unique_ptr<base::ProcessMetrics> metrics(
+      base::ProcessMetrics::CreateProcessMetrics(
+        process.Handle(), content::BrowserChildProcessHost::GetPortProvider()));
+#else
+    std::unique_ptr<base::ProcessMetrics> metrics(
+      base::ProcessMetrics::CreateProcessMetrics(process.Handle()));
+#endif
+
+    mate::Dictionary pidDict = mate::Dictionary::CreateEmpty(isolate);
+    mate::Dictionary memoryDict = mate::Dictionary::CreateEmpty(isolate);
+
+    memoryDict.Set("workingSetSize",
+            static_cast<double>(metrics->GetWorkingSetSize() >> 10));
+    memoryDict.Set("peakWorkingSetSize",
+            static_cast<double>(metrics->GetPeakWorkingSetSize() >> 10));
+
+    size_t private_bytes, shared_bytes;
+    if (metrics->GetMemoryBytes(&private_bytes, &shared_bytes)) {
+      memoryDict.Set("privateBytes", static_cast<double>(private_bytes >> 10));
+      memoryDict.Set("sharedBytes", static_cast<double>(shared_bytes >> 10));
+    }
+
+    pidDict.Set("memory", memoryDict);
+    pidDict.Set("pid", std::to_string(pid));
+    result.push_back(pidDict);
+    processEntry = process_iterator.NextProcessEntry();
+  }
+
+  return mate::ConvertToV8(isolate, result);
```

Comment:
I believe this method can just have a return type of `std::vector<mate::Dictionary>` and then you don't need to do this explicit conversion here.